tokio-threadpool/src/task/mod.rs

```diff
@@ -58,50 +60,46 @@ pub(crate) enum Run {
     Complete,
 }
 
-type BoxFuture = Box<dyn Future<Item = (), Error = ()> + Send + 'static>;
+type BoxFuture = Pin<Box<dyn Future<Output = ()> + Send + 'static>>;
 
 // ===== impl Task =====
 
 impl Task {
     /// Create a new `Task` as a harness for `future`.
     pub fn new(future: BoxFuture) -> Task {
-        // Wrap the future with an execution context.
-        let task_fut = executor::spawn(future);
-
         Task {
             state: AtomicUsize::new(State::new().into()),
             blocking: AtomicUsize::new(BlockingState::new().into()),
             next_blocking: AtomicPtr::new(ptr::null_mut()),
             reg_worker: Cell::new(None),
             reg_index: Cell::new(0),
-            future: UnsafeCell::new(Some(task_fut)),
+            future: UnsafeCell::new(Some(future)),
         }
     }
 
     /// Create a fake `Task` to be used as part of the intrusive mpsc channel
     /// algorithm.
     fn stub() -> Task {
-        let future = Box::new(futures::empty()) as BoxFuture;
-        let task_fut = executor::spawn(future);
+        let future = Box::pin(Empty) as BoxFuture;
 
         Task {
             state: AtomicUsize::new(State::stub().into()),
             blocking: AtomicUsize::new(BlockingState::new().into()),
             next_blocking: AtomicPtr::new(ptr::null_mut()),
             reg_worker: Cell::new(None),
             reg_index: Cell::new(0),
-            future: UnsafeCell::new(Some(task_fut)),
+            future: UnsafeCell::new(Some(future)),
         }
     }
 
     /// Execute the task returning `Run::Schedule` if the task needs to be
     /// scheduled again.
-    pub fn run(&self, unpark: &Arc<Notifier>) -> Run {
+    pub fn run(me: &Arc<Task>, pool: &Arc<Pool>) -> Run {
```

Comment:
nit: can this be &Arc<Self>? is that a valid receiver?

Reply:
It isnt' a valid `self`